src/path.ts

```diff
@@ -229,24 +219,10 @@ class ResourcePath extends Path {
    * @param {...string|string[]} segments - Sequence of names of the parts of
    * the path.
    */
-  constructor(projectId, databaseId, segments) {
-    segments = is.array(segments)
-      ? segments
-      : Array.prototype.slice.call(arguments, 2);
-
+  constructor(
+      private readonly projectId: string, private readonly databaseId: string,
```

Comment:
I'm personally a little 👎 on the fancy constructor assignment. @ofrobots do we have anything about this in the style guide?

Reply:
@JustinBeckwith your sensibilities are opposed to what the style guide recommends: http://go/tsstyle#parameter-properties. I'm with the style guide on this one.